src/ray/common/ray_config_def.h

```diff
@@ -768,19 +768,27 @@ RAY_CONFIG(int64_t, grpc_keepalive_time_ms, 10000)
 /// grpc keepalive timeout.
 RAY_CONFIG(int64_t, grpc_keepalive_timeout_ms, 20000)
 
-/// NOTE: we set a loose client keep alive because
+/// NOTE: We set a loose client keep alive because
 /// they have a failure model that considers network failures as component failures
 /// and this configuration break that assumption. We should apply to every other component
 /// after we change this failure assumption from code.
 /// grpc keepalive timeout for client.
 RAY_CONFIG(int64_t, grpc_client_keepalive_time_ms, 300000)
-
-/// grpc keepalive timeout for client.
+/// gRPC keep-alive timeout for client.
 RAY_CONFIG(int64_t, grpc_client_keepalive_timeout_ms, 120000)
-
+/// Number of keep-alive pings client can send before needing to send a data/header frame
+/// (0 indicates that an infinite number of pings can be sent without sending a data frame
+/// or header frame)
+RAY_CONFIG(int64_t, grpc_http2_max_pings_without_data, 0)
+/// Is it permissible to send keepalive pings from the client without any outstanding
+/// streams
+RAY_CONFIG(int64_t, grpc_keepalive_permit_without_calls, 1)
```

Comment:
I think it makes sense to override grpc_http2_max_pings_without_data to infinite, but do we need to set this to non-default? I don't see the benefit of keepalives when there are no outstanding streams and it seems like it could do harm too.

Reply:
Yeah, that one has less of an impact we can keep it as default.

What harm do you see it making though?

Reply:
Too many open connections, too many pings (if other configs are set too).